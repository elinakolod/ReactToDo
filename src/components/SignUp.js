import React, {Component} from 'react';
import {connect} from 'react-redux';
import {signUpUserRequest} from '../redux/actions';

class Signup extends Component {
  state = {
    firstName: "",
    lastName: "",
    email: "",
    password: ""
  }

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  handleSubmit = event => {
    event.preventDefault()
    this.props.signUpUserRequest(this.state)
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <h1>Sign Up For An Account</h1>

        <label>First Name</label>
        <input
          name='firstName'
          placeholder='first name'
          value={this.state.firstName}
          onChange={this.handleChange}
          /><br/>

        <label>Last Name</label>
        <input
          name='lastName'
          placeholder='last name'
          value={this.state.lastName}
          onChange={this.handleChange}
          /><br/>

        <label>Email</label>
        <textarea
          name='email'
          placeholder='email'
          value={this.state.email}
          onChange={this.handleChange}
          /><br/>

        <label>Password</label>
        <input
          type='password'
          name='password'
          placeholder='Password'
          value={this.state.password}
          onChange={this.handleChange}
          /><br/>

        <input type='submit'/>
      </form>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  signUpUserRequest: userInfo => dispatch(signUpUserRequest(userInfo))
})

export default connect(null, mapDispatchToProps)(Signup);
