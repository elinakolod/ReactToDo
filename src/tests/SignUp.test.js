import React from 'react';
import SignUp from '../components/SignUp'
import renderer from 'react-test-renderer'
import thunk from 'redux-thunk'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

const middlewares = [thunk];
const mockStore = configureStore(middlewares)
const state = {}
const store = mockStore(state)

it('matches snapshot', () => {
  const wrapper = renderer.create(<Provider store={store}><SignUp/></Provider>)

  expect(wrapper.toJSON()).toMatchSnapshot();
});
