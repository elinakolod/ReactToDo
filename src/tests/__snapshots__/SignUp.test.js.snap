// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <h1>
    Sign Up For An Account
  </h1>
  <hr />
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="formBasicFirstName"
    >
      First Name
    </label>
    <input
      className="form-control"
      id="formBasicFirstName"
      name="firstName"
      onChange={[Function]}
      placeholder="First Name"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="formBasicLastName"
    >
      Last Name
    </label>
    <input
      className="form-control"
      id="formBasicLastName"
      name="lastName"
      onChange={[Function]}
      placeholder="Last Name"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="formBasicEmail"
    >
      Email
    </label>
    <input
      className="form-control"
      id="formBasicEmail"
      name="email"
      onChange={[Function]}
      placeholder="Enter email"
      type="email"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="formBasicPassword"
    >
      Password
    </label>
    <input
      className="form-control"
      id="formBasicPassword"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    type="submit"
  >
    Submit
  </button>
</form>
`;
